include/material.hpp

```diff
@@ -116,12 +114,11 @@ struct isotropic_material {
   isotropic_material(texture_t& a)
       : albedo { a } {}
 
-  bool scatter(auto& ctx, const ray& r_in, const hit_record& rec,
-               color& attenuation, ray& scattered) const {
-    auto& rng = ctx.rng;
+  bool scatter(auto& ctx, const ray& r_in, const hit_record& rec, color& attenuation,
+               ray& scattered) const {
+    LocalPseudoRNG rng {  toseed(r_in.direction(), r_in.origin()) };
```

Comment:
```suggestion
    LocalPseudoRNG rng { toseed(r_in.direction(), r_in.origin()) };
```
I do not know why I am getting bored here...